compiler/ghc/default.nix

```diff
@@ -75,8 +84,9 @@ let
     DYNAMIC_GHC_PROGRAMS = ${if enableShared then "YES" else "NO"}
     INTEGER_LIBRARY = ${if enableIntegerSimple then "integer-simple" else "integer-gmp"}
   '' + stdenv.lib.optionalString (targetPlatform != hostPlatform) ''
-    Stage1Only = ${if targetPlatform.system == hostPlatform.system then "NO" else "YES"}
     CrossCompilePrefix = ${targetPrefix}
+  '' + stdenv.lib.optionalString (targetPlatform != hostPlatform && !(hostPlatform.isLinux && targetPlatform.isMusl)) ''
```

Comment:
Same `isCrossCompiler(stdenv)` or do we need `isCrossCompiler(targetPlatform, hostPlatform)`? I guess `stdenv` would be easier and less error prone.

Reply:
I have added isCrossHost and isCrossTarget to haskellLib